Sunrise/ViewControllers/ProductOverviewViewController.swift

```diff
@@ -0,0 +1,91 @@
+
+    var viewModel: ProductOverviewViewModel? {
+        didSet {
+            dispatch_async(dispatch_get_main_queue()) { [unowned self] in
+                self.bindViewModel()
+            }
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // After implementing POP search, view model will be instantiated with initial query
+        viewModel = ProductOverviewViewModel(limit: 15)
+    }
+
+    // MARK: - Bindings
+
+    private func bindViewModel() {
+        guard let viewModel = viewModel else { return }
+
+        navigationItem.title = viewModel.title
+
+        viewModel.isLoading.producer
+        .observeOn(UIScheduler())
+        .startWithNext({ [weak self] isLoading in
+            if !isLoading {
+                self?.collectionView?.reloadData()
+            }
+        })
+
+        viewModel.alertMessageSignal
+        .observeOn(UIScheduler())
+        .observeNext({ [weak self] alertMessage in
+            let alertController = UIAlertController(
+            title: "Oops!",
+                    message: alertMessage,
+                    preferredStyle: .Alert
+            )
+            alertController.addAction(UIAlertAction(title: "OK", style: .Cancel, handler: nil))
+            self?.presentViewController(alertController, animated: true, completion: nil)
+        })
+    }
+
+
+    // MARK: - Navigation
+
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+        if let selectedCell = sender as? ProductOverviewCell, indexPath = collectionView?.indexPathForCell(selectedCell),
+                productViewController = segue.destinationViewController as? ProductViewController, viewModel = viewModel {
+            let productDetailsViewModel = viewModel.productDetailsViewModelForProductAtIndexPath(indexPath)
+            productViewController.viewModel = productDetailsViewModel
+        }
+    }
+
+
+    // MARK: UICollectionViewDataSource
+
+    override func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let viewModel = viewModel else { return 0 }
+        return viewModel.numberOfProductsInSection(section)
+    }
+
+    override func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCellWithReuseIdentifier("ProductOverviewCell", forIndexPath: indexPath) as! ProductOverviewCell
+
+        guard let viewModel = viewModel else { return cell }
+
+        cell.productNameLabel.text = viewModel.productNameAtIndexPath(indexPath)
+        cell.priceLabel.text = viewModel.productPriceAtIndexPath(indexPath)
+
+        let priceBeforeDiscount =  NSMutableAttributedString(string: viewModel.productOldPriceAtIndexPath(indexPath))
+        priceBeforeDiscount.addAttribute(NSStrikethroughStyleAttributeName, value: 2, range: NSMakeRange(0, priceBeforeDiscount.length))
```

Comment:
I'd vote to not use extensions for classes that are not our own. I can imagine that someone wants to use parts of what is implement in Sunrise in his own app (e.g. only the cart). That would force them to have all of our extensions globally as well, since they are not namespaced.
Even worse is if two dependencies declare the same extension...